summoner-tui/src/Summoner/Tui/Validation.hs

```diff
@@ -182,6 +192,13 @@ validateKit dirs kit =
         (not $ kit ^. projectMeta . lib  || kit ^. projectMeta . exe)
         LibOrExe
 
+    validatePreludePackage :: Validation (NonEmpty FormError) ()
+    validatePreludePackage =
+      let packageName = kit ^. projectMeta . preludeName in
+      toError
+        (not $ T.null packageName || packageNameValid packageName)
+        PreludePackageValid
```

Comment:
Let's move this boolean flag into the variable like `isPreludePackageInvalid` and use it later :

```haskell
toError isPreludePackageInvalid PreludePackage
```

Reply:
Hmm, I don't think null check for package name should be externalized.

We only allow empty packages because we assume that an empty package name means "no preference". Empty package names are not valid in general.

Ideal scenario would be to make it explicit that we have either specified a custom prelude package or have not, using a sum type like:

```haskell
data PreludePreference = DefaultPrelude | CustomPrelude Text
```

Also I would prefer keeping predicates as positive (ie. using `not . isPreludePackageValid` instead of `isPreludePackageInvalid`).

What do you think?

Reply:
I find the following code more readable than the current one:

```haskell
let packageName = ...
let isPreludePackageInvalid = ...
toError isPreludePackageInvalid PreludePackage
```

It's also shorter. Because with current code `toError` application requires to split the code into multiple lines. It's hard to deduce the meaning from a bunch of boolean operators. So it makes the code easier to read if all these operators are put into a single variable with assigned meaning.

Semantically there's no difference between `isPreludePackageInvalid` and `(not isPreludePackageValid)` but if a new variable is going to be introduced, why not put this `not` application directly into variable definition :slightly_smiling_face: